src/drt/src/dr/FlexDR.cpp

```diff
@@ -2328,6 +2328,148 @@ static std::vector<FlexDR::SearchRepairArgs> strategy()
+    const auto& net = nets.at(i);
+    std::vector<gtl::polygon_90_set_data<frCoord>> routeSetByLayerNum,
+        guideSetByLayerNum;
+    routeSetByLayerNum.resize(getTech()->getLayers().size());
+    guideSetByLayerNum.resize(getTech()->getLayers().size());
+    for (const auto& shape : net->getShapes()) {
+      odb::Rect rect;
+      shape->getBBox(rect);
+      gtl::polygon_90_data<frCoord> poly;
+      vector<gtl::point_data<frCoord>> points;
+      for (auto point : rect.getPoints()) {
+        points.push_back({point.x(), point.y()});
+      }
+      poly.set(points.begin(), points.end());
+      routeSetByLayerNum[shape->getLayerNum()] += poly;
+    }
+    for (const auto& pwire : net->getPatchWires()) {
+      odb::Rect rect;
+      pwire->getBBox(rect);
+      gtl::polygon_90_data<frCoord> poly;
+      vector<gtl::point_data<frCoord>> points;
+      for (auto point : rect.getPoints()) {
+        points.push_back({point.x(), point.y()});
+      }
+      poly.set(points.begin(), points.end());
+      routeSetByLayerNum[pwire->getLayerNum()] += poly;
+    }
+    for (const auto& via : net->getVias()) {
+      {
+        odb::Rect rect;
+        via->getLayer1BBox(rect);
+        gtl::polygon_90_data<frCoord> poly;
+        vector<gtl::point_data<frCoord>> points;
+        for (auto point : rect.getPoints()) {
+          points.push_back({point.x(), point.y()});
+        }
+        poly.set(points.begin(), points.end());
+        routeSetByLayerNum[via->getViaDef()->getLayer1Num()] += poly;
+      }
+      {
+        odb::Rect rect;
+        via->getLayer2BBox(rect);
+        gtl::polygon_90_data<frCoord> poly;
+        vector<gtl::point_data<frCoord>> points;
+        for (auto point : rect.getPoints()) {
+          points.push_back({point.x(), point.y()});
+        }
+        poly.set(points.begin(), points.end());
+        routeSetByLayerNum[via->getViaDef()->getLayer2Num()] += poly;
+      }
+    }
+
+    for (const auto& shape : net->getOrigGuides()) {
+      odb::Rect rect;
+      shape.getBBox(rect);
+      gtl::polygon_90_data<frCoord> poly;
+      vector<gtl::point_data<frCoord>> points;
+      for (auto point : rect.getPoints()) {
+        points.push_back({point.x(), point.y()});
+      }
+      poly.set(points.begin(), points.end());
+      guideSetByLayerNum[shape.getLayerNum()] += poly;
+    }
+
+    for (frLayerNum lNum = 0; lNum < getTech()->getLayers().size(); lNum++) {
+      if (getTech()->getLayer(lNum)->getType() != dbTechLayerType::ROUTING
+          || lNum > TOP_ROUTING_LAYER)
+        continue;
+      if (routeSetByLayerNum[lNum].empty()) {
+        #pragma omp critical
+        netsCoverage[net.get()].push_back(-1.0);
+        continue;
+      }
+      auto routingArea = gtl::area(routeSetByLayerNum[lNum]);
+      auto coveredArea
+          = gtl::area(routeSetByLayerNum[lNum] & guideSetByLayerNum[lNum]);
+      auto coveredPercentage = (coveredArea / (routingArea * 1.0)) * 100;
```

Comment:
Though the compiler will probably optimize away '* 1.0' it would be better to just cast to double rather than multiply.